python/pyarrow/scalar.pxi

```diff
@@ -46,6 +46,9 @@ cdef class NullType(Scalar):
         """
         return None
 
+    def __eq__(self, other):
+        return NA
```

Comment:
That's not very convenient, as Python code generally expects `==` to return either True or False (yes, I know Numpy does it differently too...). @bkietz @wesm what do you think?

Reply:
That's true .. The other option is to return `False` (in any case not True), but that means that the scalar value deviates from the behaviour in the array. 

But since the scalar values are not very much developed, that's maybe not a huge problem? I also just realized that other scalar types also deviate from the array behaviour in the sense that `pa.array([1])[0] == pa.NULL` returns False and not NULL.

Reply:
If we would go with returning `pa.NULL`, we should add `__bool__` to return False, so that in a bool context like `if val1 == val2:` (where val1 and val2 could be both `pa.NULL`) this expression evaluates to False, even though the actual equality does not return False.

(and actually, we should add `__bool__` anyway, regardless of a decision on the equality, as currently `bool(pa.NULL)` is True)